src/toxcore/dht/lan_discovery.rs

```diff
@@ -0,0 +1,105 @@
+/*
+    Copyright (C) 2013 Tox project All Rights Reserved.
+    Copyright © 2018 Namsoo CHO <nscho66@gmail.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*! The inner implementation of lan discovery used only by DHT server.
+*/
+
+use toxcore::crypto_core::*;
+
+use std::net::{SocketAddr, IpAddr};
+
+use pnet_datalink;
+use ipnetwork::*;
+
+/** Structure that is needed to assemble Lan Discovery packet.
+The packet assembled is sent to Broadcast address of LAN.
+The broadcast addresses are interface broadcast and global boradcast(255.255.255.255) address for Ipv4,
+and multicast(FF02::1) address for Ipv6.
+*/
+#[derive(Debug)]
+pub struct LanDiscoveryAssembler {
```

Comment:
Name of this struct does not reflect the essence anymore (as well as docs). Do you think it's still needed at all?